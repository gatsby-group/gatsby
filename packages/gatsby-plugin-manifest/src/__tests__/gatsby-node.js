jest.mock(`fs`)

const { postBuild } = require(`../gatsby-node`)

describe(`gatsby-plugin-manifest`, () => {
  it(`should omit plugins key`, () => {
    const fsMock = jest.fn()
    require(`fs`).__setWriteFileSyncFn(fsMock)

    const manifest = {
      name: `Gatsby website`,
      short_name: `Gatsby website`,
      start_url: `/`,
      background_color: `#f7f7f7`,
      theme_color: `#191919`,
      display: `minimal-ui`,
      plugins: `here`,
    }

    postBuild(null, manifest)
    expect(fsMock.mock.calls).toMatchSnapshot()
  })

  it(`should parse keys into snake case`, () => {
    const fsMock = jest.fn()
    require(`fs`).__setWriteFileSyncFn(fsMock)

    const manifest = {
      name: `Gatsby website`,
      shortName: `Gatsby website`,
      startUrl: `/`,
      backgroundColor: `#f7f7f7`,
      themeColor: `#191919`,
      display: `minimal-ui`,
      sidebarAction: {
        defaultIcon: {
          "16": `button/geo-16.png`,
        },
      },
      plugins: `here`,
    }

    postBuild(null, manifest)
    expect(fsMock.mock.calls).toMatchSnapshot()
  })
})
