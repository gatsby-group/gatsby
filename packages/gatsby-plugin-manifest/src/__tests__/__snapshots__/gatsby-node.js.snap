// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-manifest should omit plugins key 1`] = `
Array [
  Array [
    "./public/manifest.json",
    "{\\"name\\":\\"Gatsby website\\",\\"short_name\\":\\"Gatsby website\\",\\"start_url\\":\\"/\\",\\"background_color\\":\\"#f7f7f7\\",\\"theme_color\\":\\"#191919\\",\\"display\\":\\"minimal-ui\\"}",
  ],
]
`;

exports[`gatsby-plugin-manifest should parse keys into snake case 1`] = `
Array [
  Array [
    "./public/manifest.json",
    "{\\"name\\":\\"Gatsby website\\",\\"short_name\\":\\"Gatsby website\\",\\"start_url\\":\\"/\\",\\"background_color\\":\\"#f7f7f7\\",\\"theme_color\\":\\"#191919\\",\\"display\\":\\"minimal-ui\\",\\"sidebar_action\\":{\\"default_icon\\":{\\"16\\":\\"button/geo-16.png\\"}}}",
  ],
]
`;
